system/include/compat/xmmintrin.h

```diff
@@ -16,6 +16,12 @@
 #error "SSE instruction set not enabled"
 #endif
 
+#ifdef __EMSCRIPTEN_SIMD_COMPAT_SLOW
+#define DIAGNOSE_SLOW diagnose_if(1, "Instruction emulated via slow path.", "warning")
+#else
+#define DIAGNOSE_SLOW
+#endif
+
 // Emscripten SIMD support doesn't support MMX/float32x2/__m64.
```

Comment:
Is there anything emscripten-specific about this header?  Should this say "`WebAssembly SIMD ...`..  and if so should your new macro just be `WASM_SIMD_COMPAT_SLOW` maybe?